feature-walletconnect-impl/src/main/java/jp/co/soramitsu/walletconnect/impl/presentation/requestpreview/RequestPreviewViewModel.kt

```diff
@@ -0,0 +1,206 @@
+
+        val requestedWallet = accountRepository.allMetaAccounts().firstOrNull { wallet ->
+            wallet.address(requestChain).equals(requestAddress, true)
+        }
+
+        if (requestedWallet == null) {
+            launch(Dispatchers.Main) {
+                showError(
+                    title = resourceManager.getString(R.string.common_error_general_title),
+                    message = resourceManager.getString(R.string.connection_account_not_supported_warning),
+                    positiveButtonText = resourceManager.getString(R.string.common_close),
+                    positiveClick = { walletConnectRouter.back() }
+                )
+            }
+            return@map null
+        }
+
+        val requestedWalletIcon = addressIconGenerator.createAddressIcon(
+            requestedWallet.substrateAccountId,
+            AddressIconGenerator.SIZE_BIG
+        )
+
+        WalletItemViewState(
+            id = requestedWallet.id,
+            title = requestedWallet.name,
+            isSelected = false,
+            walletIcon = requestedWalletIcon
+        )
+    }
+        .inBackground()
+        .share()
+
+    val state = combine(requestWalletItemFlow, requestChainFlow.filterNotNull(), isLoading) { requestWallet, requestChain, isLoading ->
+        val icon = GradientIconState.Remote(requestChain.icon, "EE0077")
+
+        val tableItems = listOf(
+            TitleValueViewState(
+                resourceManager.getString(R.string.common_dapp),
+                recentSession.peerMetaData?.name
+            ),
+            TitleValueViewState(
+                resourceManager.getString(R.string.common_host),
+                recentSession.peerMetaData?.dappUrl
+            ),
+            TitleValueViewState(
+                resourceManager.getString(R.string.common_network),
+                requestChain.name
+            ),
+            TitleValueViewState(
+                resourceManager.getString(R.string.common_transaction_raw_data),
+                value = "",
+                clickState = TitleValueViewState.ClickState.Value(R.drawable.ic_right_arrow_24_align_right, TRANSACTION_RAW_DATA_CLICK_ID)
+            )
+        )
+
+        requestWallet?.let {
+            RequestPreviewViewState(
+                chainIcon = icon,
+                method = recentSession.request.method,
+                tableItems = tableItems,
+                wallet = it,
+                loading = isLoading
+            )
+        } ?: RequestPreviewViewState.default
+    }
+        .stateIn(this, SharingStarted.Eagerly, RequestPreviewViewState.default)
+
+    override fun onClose() {
+        launch(Dispatchers.Main) {
+            walletConnectRouter.back()
+        }
+    }
+
+    override fun onSignClick() {
+        if (isLoading.value) return
+        val chain = requestChainFlow.value ?: return
+
+        isLoading.value = true
+        viewModelScope.launch {
+            walletConnectInteractor.onSignClick(
```

Comment:
onSignClick it would be better to name it like `handleSignAction` or something like that